File: lib/biz/spec-iot/src/test/java/site/addzero/biz/spec/iot/test/spi/TestPropertySpecProvider.java
package site.addzero.biz.spec.iot.test.spi;

import site.addzero.biz.spec.iot.IotPropertySpec;
import site.addzero.biz.spec.iot.IotThingRef;
import site.addzero.biz.spec.iot.IotValueType;
import site.addzero.biz.spec.iot.spi.IotPropertySpecProvider;

import java.util.Arrays;
import java.util.List;

public final class TestPropertySpecProvider implements IotPropertySpecProvider {

    @Override
    public String getName() {
        return "test-provider";
    }

    @Override
    public boolean supports(IotThingRef thingRef) {
        return "product".equals(thingRef.getKind()) && "demo-product".equals(thingRef.getId());
    }

    @Override
    public List<IotPropertySpec> getPropertySpecs(IotThingRef thingRef) {
        return Arrays.asList(
                IotPropertySpec.builder()
                        .identifier("temperature")
                        .name("Temperature")
                        .valueType(IotValueType.FLOAT32)
                        .build(),
                IotPropertySpec.builder()
                        .identifier("running")
                        .name("Running")
                        .valueType(IotValueType.BOOLEAN)
                        .build()
        );
    }
}

File: lib/biz/spec-iot/src/test/kotlin/site/addzero/biz/spec/iot/test/spi/TestPropertySpecProvider.kt
package site.addzero.biz.spec.iot.test.spi

import site.addzero.biz.spec.iot.IotPropertySpec
import site.addzero.biz.spec.iot.IotThingRef
import site.addzero.biz.spec.iot.IotValueType
import site.addzero.biz.spec.iot.spi.IotPropertySpecProvider

class TestPropertySpecProvider : IotPropertySpecProvider {

    override val name: String = "test-provider"

    override fun supports(thingRef: IotThingRef): Boolean {
        return thingRef.kind == "product" && thingRef.id == "demo-product"
    }

    override fun getPropertySpecs(thingRef: IotThingRef): List<IotPropertySpec> {
        return listOf(
            IotPropertySpec.builder()
                .identifier("temperature")
                .name("Temperature")
                .valueType(IotValueType.FLOAT32)
                .build(),
            IotPropertySpec.builder()
                .identifier("running")
                .name("Running")
                .valueType(IotValueType.BOOLEAN)
                .build(),
        )
    }
}
